File: app/src/main/java/com/example/calculator_0/Facade/ConverterFacade.java
package com.example.calculator_0.Facade;

import android.content.Context;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

public class ConverterFacade {
    public static void convert(EditText et, TextView tV, Spinner sp1, Spinner sp2, Context context,float[] arr){

        int first=sp1.getSelectedItemPosition();
        int second=sp2.getSelectedItemPosition();
        float input=Float.parseFloat(et.getText().toString());
        if(second==first || input==0){
            Toast toast= Toast.makeText(context,"Введенные данные некорректны!",Toast.LENGTH_LONG);
            toast.show();
        }
        else {
            float output=input/arr[second]*arr[first];
            tV.setText(String.valueOf(output));
        }
    }
}

File: app/src/main/java/com/example/calculator_0/Facade/ConverterFacade.kt
package com.example.calculator_0.Facade

import android.content.Context
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast

class ConverterFacade {
    fun convert(et: EditText, tv: TextView, sp1: Spinner,
                sp2:Spinner,context: Context,arr:FloatArray) {
        val first = sp1.selectedItemPosition
        val second = sp2.selectedItemPosition
        val input:Float = et.text.toString().toFloat()
        if(second == first || input==0F){
            Toast.makeText(context,"Введенные данные некорректны!",Toast.LENGTH_LONG).show()
        }
        else{
            tv.text = (input/arr[second]*arr[first]).toString()
        }
    }
}